### DISCUSSION TITLE
1280x720 blurry hard to read text

### DISCUSSION THREAD

**Discussion by mataski** [ihhub/fheroes2#7627] [General]
### Your question

Hi, on 1920x1080 monitor the lower resolution like e.g. 960x540x2 looks better and sharper than 1280x720? Why is that?

Problem is that 960x540x2 is too zoomed in, the 1920x1080 is too zoomed out, both are clean pixel perfect.

But 1280x720 text is distorted or something, why is that? Isn't 1280x720 a native 16:9?

E.g. 960x540x2
![image](https://github.com/ihhub/fheroes2/assets/62036912/c004fffa-603f-4ff7-a2b0-90faa89328f4)
![image](https://github.com/ihhub/fheroes2/assets/62036912/f0726615-7dbf-4673-a875-19a4dcf38ed2)

vs 1280x720
![image](https://github.com/ihhub/fheroes2/assets/62036912/38060b56-eee6-4483-85c1-223237528505)
![image](https://github.com/ihhub/fheroes2/assets/62036912/61e991d7-35af-4182-955f-ff248816a872)

Looks like other 16:9 resolutions look even worse, e.g. 1600x900
![image](https://github.com/ihhub/fheroes2/assets/62036912/90804aec-5ae5-4cc9-b6b6-2eef31cf5901)

Which is funny, we watched some YouTube videos of fdheroes2 seemed to be recorded at 720p and it looks clean and crisp.

How come lower x2 resolutions look better? Any chance to add something like x1.5? That would make 1280x720x1.5=1920x1080

Thanks.

**Comment by mataski** [comment]
I see an option with x2.2, not sure how or why this is available, but the text also looks cleaner and crisper than 1280x720

![image](https://github.com/ihhub/fheroes2/assets/62036912/1213a052-75fe-4b2c-b4b4-8d903a3cba17)

**Comment by ihhub** [comment]
The game has 2D pixel graphics. Obviously , when you use any non-integer scaling you have to blur or in other words interpolate pixels as no new pixels exist.

**Comment by ihhub** [comment]
I am making this as a discussion.

**Comment by mataski** [comment]
So there is no way to make game look clean with correct scaling with any resolution between 960x540x2 and 1920x1080?

One is too zoomed it, other is too zoomed out on 24 inch 1080p.

The goal is some resolution in the middle with correct scaling.

**Reply by Branikolog** [reply]
Hi, @mataski 

>So there is no way to make game look clean with correct scaling with any resolution between 960x540x2 and 1920x1080?

Your monitor has a fixed number of physical pixels. While running the engine in 1920x1080 each game pixel is shown with a single pixel of your monitor. In a case you set 960x540 - each game pixel is rendered with 4 physical pixels of your monitor. All other resolutions between 960 and 1920 can't render the perfectly sized image with integer number of physical pixels. Here I'm talking about running application in **fullscreen** mode. 
Running application in windowed mode engine renders each game pixel with a single monitor pixel (which makes game window really tiny for small resolutions, like 640x480), except for the (x2) resolutions (which could be rendered 2 times bigger on all resolutions up to 960x540 in your case).
x2.2 option is a minimal resolution for wide screens, which is useful mostly for mobile devices with small screens, where non-optimal scaling is barely noticeable.
So choosing resolution between fHD and 960x540 cannot create perfectly rendered pixels no matter _nearest_ or _linear_ scaling type you've chosen. But if you really want some larger visible area of the adventure map you can switch back to hardware cursor mode and select linear scaling, which brings less blurriness, than similar setup with software cursor mode (due to non-optimal scaling algorithms which are used within your OS). The resolution of your OS is being changed in this mode (hardware) and the overall picture is not that bad, but still you won't have pixel-perfect image.

**Reply by mataski** [reply]
Hi I switched to linear and applied sharpening filters with ReShade, I guess it's somewhat less blurry and playable. But still not really the same.

**Reply by apcsb** [reply]
@mataski Can you explain how you switched the scaler and applied filters? I can't find anything on it. Thanks!

**Reply by mataski** [reply]
Sorry, already beat the game, don't remember anything, lol.
